packages/dashboard/src/pages/NewWelcome/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { delay } from '@masknet/kit'
+import { DashboardRoutes, EnhanceableSite } from '@masknet/shared-base'
+import { makeStyles } from '@masknet/theme'
+import { Checkbox, FormControlLabel, Typography } from '@mui/material'
+import { memo, useCallback, useState } from 'react'
+import { useNavigate, useSearchParams } from 'react-router-dom'
+import urlcat from 'urlcat'
+import { Services } from '../../API.js'
+import { TermsAgreedContext } from '../../hooks/useTermsAgreed.js'
+import { DashboardTrans, useDashboardI18N } from '../../locales/index.js'
+import { WelcomeFrame } from '../../components/WelcomeFrame/index.js'
+import { Article } from './Article.js'
+import { SecondaryButton } from '../../components/SecondaryButton/index.js'
+import { PrimaryButton } from '../../components/PrimaryButton/index.js'
+
+const useStyles = makeStyles()((theme) => ({
+    title: {
+        fontSize: 36,
+        lineHeight: 1.2,
+        fontWeight: 700,
+    },
+    checkboxRow: {
+        marginTop: theme.spacing(1),
+        marginLeft: 0,
+    },
+    label: {
+        fontSize: 14,
+        color: theme.palette.maskColor.second,
+        fontFamily: 'PingFang SC',
+        marginRight: 0,
+        marginLeft: 0,
+    },
+    link: {
+        color: theme.palette.maskColor.main,
+        textDecoration: 'underline',
+    },
+    checkbox: {
+        padding: 0,
+        marginRight: theme.spacing(1.5),
+    },
+    buttonGroup: {
+        display: 'flex',
+        columnGap: 12,
+    },
+    policy: {
+        fontSize: 14,
+        lineHeight: '20px',
+        color: theme.palette.maskColor.second,
+        marginTop: 48,
+    },
+}))
+
+export default memo(function Welcome() {
+    const [, setAgreed] = TermsAgreedContext.useContainer()
+    const [allowedToCollect, setAllowedToCollect] = useState(true)
+    const [params] = useSearchParams()
+    const navigate = useNavigate()
+
+    const handleAgree = useCallback(async () => {
+        if (allowedToCollect) {
+            await Services.Settings.setTelemetryEnabled(true)
+        }
+        setAgreed(true)
+        const from = params.get('from')
+        const hasRedirect = from && from !== DashboardRoutes.Personas
+        if (hasRedirect) {
+            const search = params.get('search')
+            navigate(urlcat(from, search ? new URLSearchParams(search).entries() : {}))
+        }
+
+        // warm up, otherwise we can't access correct value from userGuideStatus
```

Comment:
looks strange. maybe it shouldn't be fixed in a hacky way

Reply:
Here I just copied the previous code. Can u provide the best way?